src/main/java/com/google/api/codegen/csharp/CSharpDiscoveryContext.java

```diff
@@ -170,13 +179,65 @@ public String addImport(String namespace) {
     return csharpCommon.addImport(namespace);
   }
 
-  private String packageNameAndImport(String suffix) {
-    String packageName =
-        CSharpContextCommon.s_underscoresToPascalCase(getApiaryConfig().getServiceCanonicalName());
-    String versionName = getApiaryConfig().getServiceVersion().replace('.', '_');
-    String namespace = "Google.Apis." + packageName + "." + versionName + suffix;
-    addImport(namespace);
-    return packageName;
+  // Put any using aliases at the end of other usings, with a line gap.
+  public Iterable<String> sortUsings(Iterable<String> usings) {
+    Predicate<String> isAlias =
+        new Predicate<String>() {
+          @Override
+          public boolean apply(String using) {
+            return using.contains("=");
+          }
+        };
+    FluentIterable<String> fluentUsings = FluentIterable.from(usings);
+    Iterable<String> aliases =
+        fluentUsings.anyMatch(isAlias)
+            ? FluentIterable.of(new String[] {""}).append(fluentUsings.filter(isAlias))
+            : Collections.<String>emptyList();
+    return fluentUsings
+        .filter(Predicates.<String>not(isAlias))
+        .append(aliases)
+        .transform(
+            new Function<String, String>() {
+              @Override
+              public String apply(String using) {
+                return using == "" ? "" : ("using " + using + ";");
```

Comment:
Thanks, I make this mistake so many times :(
Moved the transform, as you suggested, so the comparison no longer required.